### PR TITLE
🚀Preview Release

### LINKED ISSUE #405: 🚀Preview Release
### Pre-Release ToDo List

- [X] All issues in the assigned milestone are closed and all issue tasks are complete.
- [X] Add `🚀Preview Release` label to this issue.
- [X] Issue is assigned to a project.
- [X] Issue is assigned to a milestone.
- [x] Release PR linked to this issue.
- [x] All unit tests have been executed locally and have passed. _(Check out the appropriate release branch before running tests)_.
- [x] Version in project file updated _(All changes made directly on the release branch (preview/v#.#.#-preview.#))_.
- [x] Release notes created and added _(All changes made directly on the release branch (preview/v#.#.#-preview.#))_.
- [x] Manual QA Testing completed _(if applicable)_.
- [ ] Pull request has been approved and merged into a release branch before performing the release. _(Releases are performed on preview branches)_.
- [ ] Release to **_preview_** complete. _(The release is performed by running the `🚀Preview Release` workflow)_.

### Post-Release ToDo List

- [ ] The GitHub release has been created and is correct.
- [ ] The NuGet package has been successfully deployed to [nuget.org](https://www.nuget.org/) _(if applicable)_.
- [ ] Announcement of release on [Twitter](https://twitter.com/KDCoder) verified _(if applicable)_. _(Announcement should be performed automatically with the release)_.
- [ ] Announcement has been pushed to the [Discord](https://discord.gg/qewu6fNgv7) channel _(if applicable)_.

### Additional Information:

**_<details closed><summary>Unit Tests</summary>_**

Reasons for local unit test execution:
- Unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
- Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
</details>

---

**_<details closed><summary>Version Updating</summary>_**

The version can be updated by setting the values of the `<Version/>` and `<FileVersion/>` XML tags in the project file.
The `<Version/>` and `<FileVersion/>` values can hold production and preview releases.
The `<AssemblyVersion/>` XML tag can only hold production values.  Preview values are not allowed.

``` xml
<!--Preview Release Example-->
<Version>1.2.3-preview.4</Version>
<FileVersion>1.2.3-preview.4</FileVersion>
<AssemblyVersion>1.2.3</AssemblyVersion>
```
</details>

---

**_<details closed><summary>Release Notes</summary>_**

By default, the release notes go into a **_ReleaseNotes_** folder inside of the **_Documentation_** folder with the **_Documentation_** folder at the root of the repository.
Preview release notes path: **_./Documentation/ReleaseNotes/PreviewReleases_**.

Release note file names must follow a particular syntax and are in markdown format so they can be added to the release.

#### Release Notes File Name Syntax:
- **Preview Release Notes:**
  - **Syntax:** Release-Notes-v#.#.#-preview.#.md
  - **Example:** Release-Notes-v1.2.3-preview.4.md

Changes such as release notes and version updates should be committed to the destination branch (preview/v#.#.#-preview.#) in the pull request attached to this issue.
</details>

---

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Release Pull Request</h1>
<h2 style="font-weight:bold" align="center">📋Review Checklist📋</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

- [x] The **_base(target)_** branch for this pull request is a **_release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: release/v<major>.<minor>.<patch>
      Example: release/v1.2.3
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_release_** branch with the correct naming syntax in the following example.
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: release/v<major>.<minor>.<patch>
    Example: release/v1.2.3
    ```
  </details>

💡For more information on branching, refer to the project's [branching documentation](../../Documentation/Branching.md).

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title is set to _**🚀Preview Release**_.

- [x] A `🚀Preview Release` label has been added to this pull request. 

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] A QA issue has been created and manual QA testing has been performed for the changes being released _(if applicable)_.

### CODE REVIEW

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+As with all software, there is always a chance for issues and bugs, especially for preview releases, which is why your input is greatly appreciated. 🙏🏼
+</div>
+
+---
+
+<h2 style="font-weight:bold" align="center">New Features ✨</h2>
+
+1. [#247](https://github.com/KinsonDigital/Velaptor/issues/247) - Created a logging system for logging various information.
+   - Automatically creates 2 different types of logs in a **_logs_** directory in the same directory as the application.
+   - Standard logs for logging misc. info, warnings, and errors are logged into a text file that is prefixed with **_logs-_** and ending with a date in the format _yyyyMMdd_.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md:18]: Grammar
```suggestion
   - Standard logs for logging miscellaneous information, warnings, and errors are logged into a text file that is prefixed with **_logs-_** and end with a date in this format _yyyyMMdd_.
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+As with all software, there is always a chance for issues and bugs, especially for preview releases, which is why your input is greatly appreciated. 🙏🏼
+</div>
+
+---
+
+<h2 style="font-weight:bold" align="center">New Features ✨</h2>
+
+1. [#247](https://github.com/KinsonDigital/Velaptor/issues/247) - Created a logging system for logging various information.
+   - Automatically creates 2 different types of logs in a **_logs_** directory in the same directory as the application.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md:17]: Sentence structure
```suggestion
   - Automatically creates 2 different types of log files in a **_logs_** directory inside the same directory as the application.
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+   - Standard logs for logging misc. info, warnings, and errors are logged into a text file that is prefixed with **_logs-_** and ending with a date in the format _yyyyMMdd_.
+   - Each day will have a log file.
+   - An event log file for logging events only is logged into a text file that is prefixed with **_event-logs-_** and ends with a date with the format _yyyyMMdd_.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md:20]: sentence structure
```suggestion
   - An event log file is logged into a text file that is prefixed with **_event-logs-_** and ends with a date in this format _yyyyMMdd_.
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+   - Logging settings are set in the app settings file named **_app-settings.json_** which is located in the same directory as the application.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md:21]: sentence structure
```suggestion
   - Logging settings are located in the file named **_app-settings.json_** which is located in the application's directory.
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+   - Use the settings below to set up the type of logging behavior.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md:22]: Grammar
```suggestion
   -  Set up the type of logging behavior using the settings below:  
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+2. [#248](https://github.com/KinsonDigital/Velaptor/issues/248) - Created an application settings system.
+   - Added the ability to manage application settings.  This can also be expanded upon in the future.
+   - Added the ability to set the default window width and height when the application is executed.
+   - Added a new method overload named `CreateWindow()` with no parameters to the `App` class.  This method uses the app settings to determine the window width and height instead of 2 parameters.
+
+---
+
+<h2 style="font-weight:bold" align="center">Breaking Changes 🧨</h2>
+
+1. [#373](https://github.com/KinsonDigital/Velaptor/issues/373) - Refactored the classes below to `internal` that were not meant to be part of the public API.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md:36]: sentence structure
```suggestion
1. [#373](https://github.com/KinsonDigital/Velaptor/issues/373) - Refactored the following classes, that were not meant to be part of the public API, to `internal`: 
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+   - `ReactorUnsubscriber`
+   - `ImageService`
+   - `JSONService`
+   - `SystemMonitorService`
+2. [#373](https://github.com/KinsonDigital/Velaptor/issues/373) - Refactored the interfaces below to `internal` that were not meant to be part of the public API.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md:41]: sentence structure
```suggestion
2. [#373](https://github.com/KinsonDigital/Velaptor/issues/373) - Refactored the following interfaces, that were not meant to be part of the public API, to `internal`:
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+---
+
+<h2 style="font-weight:bold" align="center">Internal Changes ⚙️</h2>
+<h5 align="center">(Changes that do not affect users.  Not breaking changes, new features, or bug fixes.)</h5>
+
+1. [#247](https://github.com/KinsonDigital/Velaptor/issues/247) - Created the custom **_XUnit_** attributes below to create unit tests that will only run in debug or release builds.
+   - `FactForDebugAttribute`
+   - `FactForReleaseAttribute`
+2. [#247](https://github.com/KinsonDigital/Velaptor/issues/247) - Set up the **OpenGL** error callback system to take advantage of the new logging feature.
+3. [#373](https://github.com/KinsonDigital/Velaptor/issues/373) - Set many internal classes to sealed to achieve a closed by default coding practice.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.11.md:63]: Grammar
```suggestion
3. [#373](https://github.com/KinsonDigital/Velaptor/issues/373) - Set many internal classes to "sealed" in order to achieve a "closed by default" coding practice.
```

**Review by kselena — CHANGES_REQUESTED** [review]

**Review by kselena — DISMISSED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Pull request merged**